cpp/tests/io/compression_common.hpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/io/types.hpp>
+#include <cudf/utilities/error.hpp>
+
+#include <gtest/gtest.h>
+
+#include <cstdlib>
+#include <list>
+#include <string>
+#include <tuple>
+
+class tmp_env_var {
+ public:
+  explicit tmp_env_var(std::string name, std::string const& value) : name_(std::move(name))
+  {
+    auto const previous_value = std::getenv(name_.c_str());
+    if (previous_value != nullptr) { previous_value_ = std::string(previous_value); }
+
+    setenv(name_.c_str(), value.c_str(), 1);
+  }
+
+  tmp_env_var(tmp_env_var const&)            = delete;
+  tmp_env_var& operator=(tmp_env_var const&) = delete;
+  tmp_env_var(tmp_env_var&&)                 = delete;
+  tmp_env_var& operator=(tmp_env_var&&)      = delete;
```

Comment:
Deleted these to remove the sharp edges, like having a vector of `tmp_env_var` that gets resized.
If we need to copy or move these eventually, we can implement these properly.